src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php

```diff
@@ -461,6 +461,24 @@ protected function prepare_item_for_database( $request ) {
 			}
 		}
 
+		if ( ! empty( $request['author'] ) ) {
+			$post_author = (int) $request['author'];
+
+			if ( get_current_user_id() !== $post_author ) {
```

Comment:
What's this check for? Because `wp_insert_post()` will set `post_author` to `get_current_user_id()` if we don't specify?

Reply:
I lifted it directly from the rest posts controlled. 😄 

My understanding is that it's a quick way to check if the provided author value is valid. If the id matches the current user it bypasses the more expensive check a few lines below. Seems to be based on the assumption that the current user is valid, which I think is fair enough.

Reply:
This is fine as it is a copy and paste from the post controller.